llvm/include/llvm/SYCLLowerIR/CompileTimePropertiesPass.h

```diff
@@ -30,6 +30,8 @@ class IntrinsicInst;
 class CompileTimePropertiesPass
     : public PassInfoMixin<CompileTimePropertiesPass> {
 public:
+  CompileTimePropertiesPass(bool ConvertCacheControls = true)
```

Comment:
Nit; I think I would have preferred to call it something like `EarlyRun` or `EarlyPass`, so if other features are added that care whether it's transformed early or late, the bool still represents their needs.